bubblewrap.c

```diff
@@ -408,17 +421,16 @@ acquire_caps (void)
       /* Drop root uid, but retain the required permitted caps */
       if (setuid (getuid ()) < 0)
         die_with_error ("unable to drop privs");
+      did_setuid = TRUE;
     }
 
   if (is_privileged)
     {
+      uint32_t caps0 = REQUIRED_SETUP_CAPS_0 | requested_caps[0];
```

Comment:
is_privileged means "we're not using user namespaces" in the code, not that we're initially launched by root.
The reason its doing this check instead of comparing effective uid is to also support filesystem caps instead of setuid.